File: 2270.Number-of-Ways-to-Split-Array/2270.number-of-ways-to-split-array.py
#
# @lc app=leetcode id=2270 lang=python
#
# [2270] Number of Ways to Split Array
#

# @lc code=start
class Solution(object):
    def waysToSplitArray(self, nums):
        ans = 0 #記錄
        left = [nums[0]]
        right = nums [1:]
        left_sum = sum(left)
        right_sum = sum(right)
        for i in range(len(nums)-1):
            if left_sum >= right_sum:
                ans += 1
            left_sum += right[i]
            right_sum -= right[i]
        return ans
        """
        :type nums: List[int]
        :rtype: int
        """
        
# @lc code=end

test = Solution()
print(test.waysToSplitArray([2,3,1,0]))

File: 2270.Number-of-Ways-to-Split-Array/2270.number-of-ways-to-split-array.cpp
/*
 * @lc app=leetcode id=2270 lang=cpp
 *
 * [2270] Number of Ways to Split Array
 */

// @lc code=start
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>

using namespace std;

class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int ans = 0;
        long long int left_sum = nums[0];
        long long int right_sum = 0;
        for (int i = 1 ; i<nums.size() ; i++){
            right_sum += nums[i];
        }
        for (int i=1; i<nums.size() ;i++){
            if (left_sum >= right_sum){
                ans += 1;
            }
            left_sum += nums[i];
            right_sum -= nums[i];
        }
        return ans;
    }
};
// @lc code=end

